.github/workflows/golang.yml

```diff
@@ -30,14 +30,18 @@ jobs:
 
   end-to-end:
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        fabric_version: [2.4.8, 2.5.0-beta]
     steps:
       - uses: actions/checkout@v3
       - uses: actions/setup-go@v3
         with:
           go-version: '1.19'
-      - run: curl --location https://raw.githubusercontent.com/hyperledger/fabric/main/scripts/install-fabric.sh | bash -s
-        name: download fabric sample following official guide
-      # - run: ./network.sh up
+      - run: curl -sSLO https://raw.githubusercontent.com/hyperledger/fabric/main/scripts/install-fabric.sh && chmod +x install-fabric.sh
+        name: download fabric install script following official guide
+      - run: ./install-fabric.sh --fabric-version ${{ matrix.fabric_version }}
+        name: download fabric following official guide
       - run: ./network.sh up createChannel -c mychannel
```

Comment:
should we add create channel here?

Reply:
I had to change the setup to have network.sh create the channel since the channel config API steps did not configure anchor peers, making the channel unusable with service discovery. See #87

Reply:
We could:

1. Enhance the channel config API (or its usage in the tests) to configure anchor peers, then have that create the channel; or
2. Have one e2e test for chaincode deployment that uses network.sh to create the channel, and a second e2e test that tests the channel config API

Reply:
I suppose we'd better to take 1st option in another PR. so that with admin sdk, user able to set up fabric network from channel creation.

Reply:
Definitely the channel config API needs to be able to easily configure anchor peers. The question is just whether:

1. the e2e test for chaincode deployment uses the channel config API to create the channel; or
2. there are separate e2e tests for channel config and chaincode deployment

Reply:
I suppose the 1st approach matches a long term goal, if we considering we may decouple peer client features from peer command. Hence ... just imaging a binary based on admin-sdk, and fabric user will (as a example use case)
- download fabric sample
- download binary based admin-sdk
- create channel
- create/deploy chaincode
- start tx....

hence as a long term goal, a separate e2e test for chaincode deployment will depends on the e2e test case for  channel config.
Thus, I suppose we don't need separate e2e tests aren't we?

Reply:
Let me try if I can make it in this PR.

Reply:
let me try if I can make it with this pr

Reply:
I am fine with either approach. The issue is that right now channel config is no longer being tested. If you can add anchor peer configuration then having one e2e test is fine. If not, it might be worth having two e2e tests so we are getting coverage of both.

At some point it _might_ be useful to test combinations, like old-style channel config with SDK chaincode deployment, and SDK channel config with old-style chaincode deployment, just to make sure we have compatibility with the previous approach. I'm not 100% sure it's worth the time and effort, but maybe.

Reply:
I got your point, it's great. let's try it out.